### ISSUE TITLE
HLint config in subfolder not picked up

### ISSUE THREAD

**Issue by georgefst** [haskell/haskell-language-server#1260]
haskell-language-server version: 0.8.0.0 (GHC: 8.10.2) (PATH: /home/gthomas/.cabal/bin/haskell-language-server)
Tool versions found on the $PATH
cabal:          3.2.0.0
stack:          Not found
ghc:            8.10.2
VScode

My workspace folder has a `haskell` subfolder, but `.hlint.yaml` is only picked up if I move it to the very top level of the project, which is inconvenient. Everything else works fine.

**Comment by jneira** [comment]
Only to confirm, if you execute hlint in the cli, cd'ing in the workspace folder, does it work?

**Comment by georgefst** [comment]
Yes. HLS even works if I open just the subfolder using `Add Folder to Workspace`.

**Comment by georgefst** [comment]
Possibly relatedly, I have essentially the same issue with GHC environment files (they have to be in the root workspace directory, not just the one containing the relevant source file).

Perhaps I should just be using workspaces differently... But it does at least seem like a gotcha worth documenting somewhere - it's confusing when something can be loaded by `ghc`/`ghci` but HLS complains about not finding packages.

**Comment by alanz** [comment]
For the GHC source I just symlink the hlint config  to the top directory. FYI.
